### PR TITLE
Tools fixes

### PR DESCRIPTION
**Pull request by iulianpascalau**
- made the trie checker able to check one directory only
- made the general DB merger check for an empty destination directory

### CODE REVIEW

File: trieTools/trieChecker/README.md [hunk 1]
```diff
@@ -6,5 +6,5 @@ This tool is used to check the correct loading of a state trie. Works with user
 # How to use
 
 1. compile the binary by issuing a `go build` command in elrond-tools-go/trieTools/trieChecker directory
-2. create a `db` directory and place inside directories `0`, `1` ... that contains the state data
+2. create a `db` directory and place inside directories `0`, `1` ... that contains the state data, alternatively, you can place a random named directory and use that solely to load the data
```
Comment (andreibancioiu) [inline trieTools/trieChecker/README.md:9]: `randomly` (suggestion)

File: trieTools/trieToolsCommon/common.go [hunk 2]
```diff
@@ -146,12 +149,33 @@ func GetTrie(flags ContextFlagsConfig, maxDBValue int) (common.Trie, error) {
...
+		BatchDelaySeconds: dbConfig.BatchDelaySeconds,
+		MaxBatchSize:      dbConfig.MaxBatchSize,
+		MaxOpenFiles:      dbConfig.MaxOpenFiles,
+	}
+
+	return storageUnit.NewStorageUnitFromConf(cacheConfig, dbConf)
+}
+
+// GetTrie will create and return a trie using the provided flags
+func GetTrie(storer storage.Storer) (common.Trie, error) {
```
Comment (andreibancioiu) [inline trieTools/trieToolsCommon/common.go:173]: `GetTrie()` and `GetStorer()` create objects, but that should be all right in this context.

**Review by andreibancioiu — DISMISSED** [review]

File: trieTools/trieToolsCommon/common.go (hunk 2 above)
Comment (iulianpascalau) [reply trieTools/trieToolsCommon/common.go:173]: Renamed them to Create.... as they better suggest what are they doing. Thanks

File: trieTools/trieChecker/README.md (hunk 1 above)
Comment (iulianpascalau) [reply trieTools/trieChecker/README.md:9]: changed

File: dbMerger/path/osOperationsHandler_test.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+		err = ioutil.WriteFile(path.Join(workingDir, "test.log"), []byte("data"), os.ModePerm)
+		assert.Nil(t, err)
+
+		err = handler.CheckIfDirectoryIsEmpty(workingDir)
+		assert.NotNil(t, err)
+		assert.True(t, errors.Is(err, errDirectoryIsNotEmpty))
+	})
+}
+
+func readFileContent(tb testing.TB, path string) string {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply dbMerger/path/osOperationsHandler_test.go:94]: switched to use the io.ReadAll function

**Review by schimih — DISMISSED** [review]
It was a bit confusing for me to see that 
`getPruningStorer(..)(storage.Storer, error)`
`getDB(..)(storage.Storer, error)`
both are returning basically a storer as a db. (is just a naming stuff)

**Comment by iulianpascalau** [comment]
> It was a bit confusing for me to see that `getPruningStorer(..)(storage.Storer, error)` `getDB(..)(storage.Storer, error)` both are returning basically a storer as a db. (is just a naming stuff)

the `getDB` function acts like a factory method, basically changing the calls based on some condition(s). Renamed to `createDB` as to better express what the method should do

**Review by andreibancioiu — APPROVED** [review]

**Review by schimih — APPROVED** [review]

**Review by schimih — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**